packages/host/tests/integration/components/ai-assistant-panel-test.gts

```diff
@@ -1327,6 +1328,196 @@ module('Integration | ai-assistant-panel', function (hooks) {
+      'm.relates_to': {
+        rel_type: 'm.replace',
+        event_id: eventId,
+      },
+    });
+    await waitUntil(() => {
+      const element = document.querySelector(
+        '[data-test-message-idx="0"] details[data-test-reasoning]',
+      );
+      return element?.textContent?.includes?.(
+        'they want to know what kind of dog to get',
+      );
+    });
+    assert
+      .dom('[data-test-message-idx="0"] details[data-test-reasoning]')
+      .containsText('they want to know what kind of dog to get');
+    assert
+      .dom('[data-test-message-idx="0"] details[data-test-reasoning]')
+      .hasAttribute('open');
+
+    await click(
+      '[data-test-message-idx="0"] details[data-test-reasoning] summary',
+    );
+    await waitFor(
+      `[data-test-message-idx="0"] details[data-test-reasoning]:not([open])`,
+    );
+    assert
+      .dom('[data-test-message-idx="0"] details[data-test-reasoning]')
+      .doesNotHaveAttribute('open');
+
+    await simulateRemoteMessage(roomId, '@aibot:localhost', {
+      [APP_BOXEL_REASONING_CONTENT_KEY]:
+        'OK, they want to know what kind of dog to get. Let me think about what relevant details I know about them.\n\nThey like beagles.',
+      body: null,
+      msgtype: 'm.text',
+      formatted_body: null,
+      isStreamingFinished: false,
+      'm.relates_to': {
+        rel_type: 'm.replace',
+        event_id: eventId,
+      },
+    });
+    assert
+      .dom('[data-test-message-idx="0"] details[data-test-reasoning]')
+      .doesNotHaveAttribute('open');
+
+    await simulateRemoteMessage(roomId, '@aibot:localhost', {
+      [APP_BOXEL_REASONING_CONTENT_KEY]:
+        'OK, they want to know what kind of dog to get. Let me think about what relevant details I know about them.\n\nThey like beagles.',
+      body: 'You should get a beagle.',
+      msgtype: 'm.text',
+      formatted_body: 'You should get a beagle.',
+      isStreamingFinished: false,
+    });
+    assert
+      .dom('[data-test-message-idx="0"] details[data-test-reasoning]')
+      .doesNotHaveAttribute('open');
+    await click(
+      '[data-test-message-idx="0"] details[data-test-reasoning] summary',
+    );
+    assert
+      .dom('[data-test-message-idx="0"] details[data-test-reasoning]')
+      .hasAttribute('open');
+
+    await simulateRemoteMessage(roomId, '@aibot:localhost', {
+      [APP_BOXEL_REASONING_CONTENT_KEY]:
+        'OK, they want to know what kind of dog to get. Let me think about what relevant details I know about them.\n\nThey like beagles.',
+      body: 'You should get a beagle. They are great companions.',
+      msgtype: 'm.text',
+      formatted_body: 'You should get a beagle. They are great companions.',
+      isStreamingFinished: false,
+      'm.relates_to': {
+        rel_type: 'm.replace',
+        event_id: eventId,
+      },
+    });
+    assert
+      .dom('[data-test-message-idx="0"] details[data-test-reasoning]')
+      .hasAttribute('open');
+  });
+
```

Comment:
Maybe it’s unorthodox for an integration test but this brings the file over 3000 characters, I think it’s time to break it up.

Reply:
Yikes. Agreed.